Unity-Technologies/ScriptableRenderPipeline
# Prototype for stereo support in compute and non-compute shaders

### Purpose of this PR
Why is this PR needed, what hard problem is it solving/fixing?
- Prototype fix for stereo screen-space shadows 
- Prototype fix for stereo camera motion vectors (contains changeset from this PR: https://github.com/Unity-Technologies/ScriptableRenderPipeline/pull/1933)
The completed fix for screen-space shadows will be slightly more involved so this seems like a good "save point".

---
### Release Notes
Please add any useful notes about the feature/fix that might be helpful for others.
Please also review this PR as a representative of the compute shader strategy for stereo rendering: in C#, I added code to dispatch the screenspaceshadow.compute shader once per-eye, with the appropriate matrices attached. 

---
### Testing status
**Katana Tests**: First off we need to make sure the Katana SRP tests are green?
https://katana.bf.unity3d.com/projects/com.unity.render-pipelines/builders?automation-tools_branch=add-platform-filter&ScriptableRenderLoop_branch=hdrp_xr&unity_branch=trunk

**Manual Tests**: What did you do?
Enabled VR and ran with OpenVR SDK + HTC Vive. 

**Automated Tests**: What did you setup?
Nothing special- just wanted to make sure existing tests didn't break.

Any test projects to go with this to help reviewers?
I can provide my project upon request.

---
### Overall Product Risks
**Technical Risk**: None, Low, Medium, High?
Low

**Halo Effect**: None, Low, Medium, High?
Low- higher impact changes are coming soon, which is why I wanted to get these changes in first.

---
### Comments to reviewers
Will wait for MSAA PR to complete to ensure these changes merge nicely (https://github.com/Unity-Technologies/ScriptableRenderPipeline/pull/1916)
